data/logs_model/logs_producer/splunk_logs_producer.py

```diff
@@ -22,19 +23,50 @@ def __init__(
         url_scheme="https",
         verify_ssl=True,
         index_prefix=None,
+        ssl_ca_path=None,
     ):
+        connect_args = {
+            "host": host,
+            "port": port,
+            "token": bearer_token,
+            "scheme": url_scheme,
+            "verify": verify_ssl,
+        }
+        if ssl_ca_path:
+            # Create an SSLContext object for cert validation
+            context = ssl.create_default_context()
+            context.check_hostname = True
+            context.verify_mode = ssl.CERT_REQUIRED
+            if not verify_ssl:
+                context.check_hostname = False
+                context.verify_mode = ssl.CERT_NONE
+            try:
+                context.load_cert_chain(certfile=ssl_ca_path)
+            except ssl.SSLError as e:
+                logger.exception("SSL certificate error %s", e)
+                raise e
+            except FileNotFoundError as e:
+                logger.exception("Path to cert file is not valid %s", e)
+                raise e
+            connect_args["context"] = context
+        elif verify_ssl:
+            raise ValueError("SSL cert path needs to be specified")
+
         try:
-            service = client.connect(
-                host=host, port=port, token=bearer_token, scheme=url_scheme, verify=verify_ssl
-            )
-        except Exception as ex:
-            logger.exception("Failed to connect to Splunk instance %s", ex)
-            raise ex
+            service = client.connect(**connect_args)
+        except Exception as e:
+            logger.exception("Failed to connect to Splunk instance %s", e)
+            raise e
+
         try:
             self.index = service.indexes[index_prefix]
             logger.info("splunk index %s", self.index)
         except KeyError:
             self.index = service.indexes.create(index_prefix)
+        except ConnectionRefusedError as e:
+            raise Exception(
+                "Connection to Splunk refused, check if Splunk instance is available %s", e
+            )
```

Comment:
You replace a detailed exception with a generic one, that caller won't be able to distinguish it from any other error. Why do you want to do it?